### PR TITLE
feat: validate SSZ schemas for containers

### LINKED ISSUE #875: Validate container schemas at compile-time
This includes two things:
- add some public function `SszEx.validate_schema!(schema)`, that raises an error in case the schema is invalid
- use that function for validating `Container` schemas

### PR DESCRIPTION
**Pull request by MegaRedHand**
Closes #875

### CODE REVIEW

**Comment by mpaulucci** [comment]
Im curious about what bug did this catch if there was any?

File: lib/types/beacon_chain/beacon_block_body_deneb.ex [hunk 1]
```diff
@@ -63,7 +63,8 @@ defmodule Types.BeaconBlockBodyDeneb do
       {:execution_payload, Types.ExecutionPayloadDeneb},
       {:bls_to_execution_changes,
        {:list, Types.BLSToExecutionChange, ChainSpec.get("MAX_BLS_TO_EXECUTION_CHANGES")}},
-      {:blob_kzg_commitments, {:list, TypeAliases.kzg_commitment()}}
+      {:blob_kzg_commitments,
+       {:list, TypeAliases.kzg_commitment(), ChainSpec.get("MAX_BLOB_COMMITMENTS_PER_BLOCK")}}
```
Comment (MegaRedHand) [inline lib/types/beacon_chain/beacon_block_body_deneb.ex:67]: This one

Comment (MegaRedHand) [reply lib/types/beacon_chain/beacon_block_body_deneb.ex:67]: This is also extensible, so we could potentially add more checks in the future.

Comment (mpaulucci) [reply lib/types/beacon_chain/beacon_block_body_deneb.ex:67]: super cool!

**Review by mpaulucci — APPROVED** [review]

**Pull request merged**